CFD-294 : Homepage : My Groups (Logged-in users).

nader77 opened the pull request:
#478

nader77 commented:
Update:
Added all the functionality for my groups block.
Displays 3 groups in alphabetical order, The rest can be view when clicking on "Show all" button.

Custom view for this block displays:
* group-type icon
* group name (clickable, link to group dashboard)
* group owner (clickable, link to public person profile)
* number of group members

As specified in the Wireframe. @HelenaEksler You should use the same node view for "Suggested groups" in PR #480 (Both blocks should have the same view).

nader77 commented:
Block collapsed: 
![Uploading Selection_046.png . . .]()

Block expanded:
![selection_047](https://cloud.githubusercontent.com/assets/7369740/6203685/08069106-b537-11e4-82f4-23190be38eea.png)

There's no content type icon for groups in the theme.

amitaibu commented on capacity4more/modules/c4m/ds/c4m_ds/includes/c4m_ds.theme.node.inc:
```diff
@@ -276,6 +276,30 @@ function theme_c4m_ds_group_access_info($field) {
+
+  $tag['element'] = array(
+    '#tag'   => 'span',
+    '#value' => $group_members . '&nbsp;' . t('members'),
```
move everything inside the t() --  t('@count members')

amitaibu commented on capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc:
```diff
@@ -0,0 +1,86 @@
+  $extra_groups = '';
+
+  // Get All user's groups (We get them sorted by id).
+  $groups = og_get_groups_by_user($user);
```
I think you can just ``og_get_groups_by_user()``

nader77 commented:
Update:
* Fixed syntax comments.
* Fixed the homepage layout.

My groups and homepage layout:
![groups-2015-02-16_11](https://cloud.githubusercontent.com/assets/7369740/6209628/7b001170-b5d0-11e4-90df-74e576178713.gif)

nader77 commented:
@HelenaEksler PR is ready for review.

HelenaEksler commented:
@nader77  Looks fine. As soon as tests are green - assign to Peter.

amitaibu commented on capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc:
```diff
@@ -0,0 +1,85 @@
+  }
+
+  // Load node for each group and add the first 2 letters of the group title as a key.
+  foreach ($groups['node'] as $group) {
```
better do a ``node_load_multiple($groups['node'])``

amitaibu replied:
better do a ``node_load_multiple($groups['node'])``

nader77 commented:
@amitaibu @HelenaEksler Fixed.

zero2one commented:
@nader77 
When you login as the site administrator (admin): 

There is a difference between the number of groups on the profile page (8):
![c4m-profile-groups](https://cloud.githubusercontent.com/assets/133124/6213615/f06fe108-b5f1-11e4-85cc-350f76db7b2e.png)

And the number of groups on the site homepage (6):
![c4m-homepage-groups](https://cloud.githubusercontent.com/assets/133124/6213616/f35fd6f2-b5f1-11e4-952b-c9ea139e803d.png)

nader77 commented:
@zero2one The other 2 are Projects, They should be displayed here ?

zero2one commented:
@nader77 Ouch, sorry, did not notice it…

You are correct, there should only be groups in the list.

Can you do a re-roll?

nader77 commented:
@zero2one Re-rolled.

zero2one commented:
Seems that every pulled-in pull request blocks the other pull requests about the home page…

@nader77  Can you re-roll please?

HelenaEksler commented:
@zero2one Re-roled

zero2one commented:
Needs a follow up for theming.
